app/services/hearing_recorder.rb

```diff
@@ -1,17 +1,17 @@
 # frozen_string_literal: true
 
 class HearingRecorder < ApplicationService
-  def initialize(hearing_id)
+  def initialize(hearing_id, body = nil)
     @hearing = Hearing.find_or_initialize_by(id: hearing_id)
+    @body = body.presence || HearingFetcher.call(hearing.id).body
```

Comment:
Seems this is a good point to extract `HearingFetcher` call outside this class and inject the body either coming from the `HearingUpdated` update from HMCTS or through `HearingFetcher` call.

Should we invert the dependency between these two services so we call `HearingRecorder` from within `HearingFetcher` call? 
In the near future probably we will have a controller action calling both `HearingFetcher` and `HearingRecorder` together so the services are not coupled between them. But as a middle step inverting the dependency so `HearingRecorder` codebase does not have conditional flows and gets called the same from both `HearingFetcher` and from the `HearingsController` seems a good compromise meanwhile.

Let me know your thoughts on this :)

Reply:
Definitely agree on that, it makes sense to invert them. This version doesn't make sense with the POST endpoint addition.